concordium-contracts-common/src/types.rs

```diff
@@ -1406,6 +1552,94 @@ pub struct Policy<Attributes> {
     pub items:             Attributes,
 }
 
+/// Generate a vector of random key-value pairs with no duplication
+/// The length of the resulting vector is <= `size`
+#[cfg(feature = "concordium-quickcheck")]
+fn gen_no_dup_kv_vec<A: quickcheck::Arbitrary + Ord, B: quickcheck::Arbitrary>(
+    g: &mut Gen,
+    size: usize,
+) -> Vec<(A, B)> {
+    use std::collections::BTreeMap;
+    let mut m: BTreeMap<A, B> = BTreeMap::new();
+    for _ in 0..size {
+        let k = A::arbitrary(g);
+        let v = B::arbitrary(g);
+        m.insert(k, v);
+    }
+    m.into_iter().collect()
+}
+
+/// Generate a random `u64` value in the given range by shifting a random `u64`
+/// value using the `%` operator. The reason for doing it this way is that the
+/// range generation method is not exposed by QuickCheck's `Gen`.
+#[cfg(feature = "concordium-quickcheck")]
+fn gen_range_u64(g: &mut Gen, range: core::ops::Range<u64>) -> u64 {
+    let i: u64 = quickcheck::Arbitrary::arbitrary(g);
+    i % (range.end - range.start) + range.start
+}
+
+/// Generate a random `u8` value in the given range by shifting a random `u8`
+/// value using the `%` operator. The reason for doing it this way is that the
+/// range generation method is not exposed by QuickCheck's `Gen`.
+#[cfg(feature = "concordium-quickcheck")]
+fn gen_range_u8(g: &mut Gen, range: core::ops::Range<u8>) -> u8 {
+    let i: u8 = quickcheck::Arbitrary::arbitrary(g);
+    i % (range.end - range.start) + range.start
+}
+
+#[cfg(feature = "concordium-quickcheck")]
+fn valid_owned_policy(op: &OwnedPolicy) -> bool {
```

Comment:
Please document this function with a one liner.

Reply:
Done.